x/gamm/types/errors.go

```diff
@@ -29,19 +29,18 @@ var (
 	ErrDenomNotFoundInPool = sdkerrors.Register(ModuleName, 11, "denom does not exist in pool")
 	ErrDenomAlreadyInPool  = sdkerrors.Register(ModuleName, 12, "denom already exists in the pool")
 
-	ErrEmptyRoutes              = sdkerrors.Register(ModuleName, 21, "routes not defined")
-	ErrEmptyPoolAssets          = sdkerrors.Register(ModuleName, 22, "PoolAssets not defined")
-	ErrNegativeSwapFee          = sdkerrors.Register(ModuleName, 23, "swap fee is negative")
-	ErrNegativeExitFee          = sdkerrors.Register(ModuleName, 24, "exit fee is negative")
-	ErrTooMuchSwapFee           = sdkerrors.Register(ModuleName, 25, "swap fee should be lesser than 1 (100%)")
-	ErrTooMuchExitFee           = sdkerrors.Register(ModuleName, 26, "exit fee should be lesser than 1 (100%)")
-	ErrNotPositiveWeight        = sdkerrors.Register(ModuleName, 27, "token weight should be greater than 0")
-	ErrWeightTooLarge           = sdkerrors.Register(ModuleName, 28, "user specified token weight should be less than 2^20")
-	ErrNotPositiveCriteria      = sdkerrors.Register(ModuleName, 29, "min out amount or max in amount should be positive")
-	ErrNotPositiveRequireAmount = sdkerrors.Register(ModuleName, 30, "required amount should be positive")
-	ErrTooManyTokensOut         = sdkerrors.Register(ModuleName, 31, "tx is trying to get more tokens out of the pool than exist")
-	ErrSpotPriceOverflow        = sdkerrors.Register(ModuleName, 32, "invalid spot price (overflowed)")
-	ErrSpotPriceInternal        = sdkerrors.Register(ModuleName, 33, "internal spot price error")
+	ErrEmptyPoolAssets          = sdkerrors.Register(ModuleName, 21, "PoolAssets not defined")
+	ErrNegativeSwapFee          = sdkerrors.Register(ModuleName, 22, "swap fee is negative")
+	ErrNegativeExitFee          = sdkerrors.Register(ModuleName, 23, "exit fee is negative")
+	ErrTooMuchSwapFee           = sdkerrors.Register(ModuleName, 24, "swap fee should be lesser than 1 (100%)")
+	ErrTooMuchExitFee           = sdkerrors.Register(ModuleName, 25, "exit fee should be lesser than 1 (100%)")
+	ErrNotPositiveWeight        = sdkerrors.Register(ModuleName, 26, "token weight should be greater than 0")
+	ErrWeightTooLarge           = sdkerrors.Register(ModuleName, 27, "user specified token weight should be less than 2^20")
+	ErrNotPositiveCriteria      = sdkerrors.Register(ModuleName, 28, "min out amount or max in amount should be positive")
+	ErrNotPositiveRequireAmount = sdkerrors.Register(ModuleName, 29, "required amount should be positive")
+	ErrTooManyTokensOut         = sdkerrors.Register(ModuleName, 30, "tx is trying to get more tokens out of the pool than exist")
+	ErrSpotPriceOverflow        = sdkerrors.Register(ModuleName, 31, "invalid spot price (overflowed)")
+	ErrSpotPriceInternal        = sdkerrors.Register(ModuleName, 32, "internal spot price error")
```

Comment:
It seems like we are assigning errors to previously assigned error numbers, which AFAIK we should not do

Reply:
I think this concern is valid across patches due to state-compatibility concerns.

Since this is a major upgrade, should be fine. 

I will ask one more reviewer to confirm if that's the case

Reply:
I prefer we don't do any more sdkerrors.register anywhere in the codebase.

Reply:
Yeah I only removed an sdk error here and ported it as Go error to the new `swaprouter` module. I did not add any new sdkerror.

However, removing an sdkerror with an ID in the middle, made me have to lower all remaining IDs by 1.

I'm wondering if that is allowed?